src/app/clusters/mode-select-server/mode-select-server.cpp

```diff
@@ -103,7 +113,119 @@ bool emberAfModeSelectClusterChangeToModeCallback(CommandHandler * commandHandle
     return true;
 }
 
+/**
+ * Callback for Mode Select Cluster Server Initialization.
+ * Enabled in src/app/zap-templates/templates/app/helper.js
+ * @param endpointId    id of the endpoint that is being initialized
+ */
+void emberAfModeSelectClusterServerInitCallback(EndpointId endpointId)
+{
+#ifndef IGNORE_MODE_SELECT_CLUSTER_START_UP_MODE
+    // StartUp behavior relies on CurrentMode StartUpMode attributes being non-volatile.
+    if (areStartUpModeServerAttributesNonVolatile(endpointId))
+    {
+        // Read the StartUpMode attribute and set the CurrentMode attribute
+        // The StartUpMode attribute SHALL define the desired startup behavior of a
+        // device when it is supplied with power and this state SHALL be
+        // reflected in the CurrentMode attribute.  The values of the StartUpMode
+        // attribute are listed below.
+
+        // Initialize startUpOnOff to max UINT8
+        uint8_t startUpMode = 0xFF;
+        EmberAfStatus status = Attributes::StartUpMode::Get(endpointId, &startUpMode);
+        if (status == EMBER_ZCL_STATUS_SUCCESS)
+        {
+            // Initialise currentMode to 0
+            uint8_t currentMode = 0;
+            status            = Attributes::CurrentMode::Get(endpointId, &currentMode);
+            if (status == EMBER_ZCL_STATUS_SUCCESS)
+            {
+                if (startUpMode != currentMode) {
+                    status = Attributes::CurrentMode::Set(endpointId, startUpMode);
```

Comment:
How does this interact with OnMode?

Reply:
OnMode will have priority over this, as per the spec
> If this attribute is not null, the CurrentMode attribute SHALL be set to the StartUpMode value, when the server is powered up, except in the case when the OnMode attribute overrides the StartUpMode attribute.

Will be implemented in another PR

Reply:
> Will be implemented in another PR

Please make sure there's an issue filed tracking that.

Reply:
We plan to reference this same issue. Once both this PR and this https://github.com/project-chip/connectedhomeip/pull/15888 have been merged.

Reply:
The point is, please make sure the issue is filed so it does not fall through the cracks.

Reply:
Created new issue to track this: https://github.com/project-chip/connectedhomeip/issues/16469